### PR TITLE
fix: add AMI as clustering metric

### PR DESCRIPTION
**Pull request by lironsoffer**
## Summary

Adds [Adjusted Mutual Information (AMI)](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.adjusted_mutual_info_score.html) as a clustering metric alongside the existing V-measure in `AbsTaskClustering`. Both metrics are now computed on every bootstrap iteration — no opt-in.

Inside `_evaluate_clustering_bootstrapped` a small `metric_funcs` mapping pairs metric names with their sklearn implementations; the bootstrap loop iterates over it, and the result dict downstream is built from the same keys — so adding another metric is a one-line change.

## Motivation

A reviewer on PR #4609 noted that the paper introducing the HumanConceptsClustering dataset ([Shani et al. 2025, arXiv:2505.17117](https://arxiv.org/abs/2505.17117)) uses **AMI** as its main metric and suggested MTEB should support it. AMI is the standard choice when comparing clusterings with chance correction.

PR #4609 will then set `main_score = "ami"` once this lands.

## Backward compatibility

- The result dict keeps the `v_measures` / `v_measure` / `v_measure_std` keys exactly as before.
- New keys: `ami_scores`, `ami`, `ami_std`.
- One extra scoring call per bootstrap iteration; the k-means fit (which dominates cost) is unchanged.

## Test plan

- [x] `make lint` passes
- [x] `pytest tests/ -k clustering -x` — 220 passed, 3 skipped (gradio/audio not installed), 0 failed
- [x] `make typecheck` clean for `mteb/abstasks/clustering.py` (pre-existing errors in unrelated files: `torchcodec.decoders`, `gradio`)

### CODE REVIEW

**Review by Samoed — COMMENTED** [review]
I don't think this is necessary to make this metric `opt-in`. You can measure it always, just as a new meetric

**Comment by lironsoffer** [comment]
Added some tests as well, not sure if needed but just in case

**Comment by Samoed** [comment]
Example run:
```json
{
  "dataset_revision": "24640382cdbf8abc73003fb0fa6d111a705499eb",
  "task_name": "RedditClustering.v2",
  "mteb_version": "2.12.25",
  "scores": {
    "test": [
      {
        "v_measures": {
          "Level 0": [
            0.410996,
            0.412815,
            0.40289,
            0.407631,
            0.438605,
            0.409442,
            0.43209,
            0.406192,
            0.430138,
            0.425363
          ]
        },
        "v_measure": 0.417616,
        "v_measure_std": 0.012031,
        "ami_scores": {
          "Level 0": [
            0.398891,
            0.400919,
            0.39046,
            0.395288,
            0.426861,
            0.397249,
            0.420219,
            0.393739,
            0.418183,
            0.413349
          ]
        },
        "ami": 0.405516,
        "ami_std": 0.012235,
        "main_score": 0.417616,
        "hf_subset": "default",
        "languages": [
          "eng-Latn"
        ],
        "mteb_version": "2.12.25"
      }
    ]
  },
  "evaluation_time": 1.522629976272583,
  "kg_co2_emissions": null,
  "date": 1778504977.407689
}
```

**Review by Samoed — APPROVED** [review]
Looks good for me

File: tests/test_abstasks/test_clustering_metrics.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            max_depth=None,
+            rng_state=random.Random(0),
+            seed=0,
+        )
+
+        mean_ami = float(np.mean(scores["ami"]["Level 0"]))
+        assert abs(mean_ami) < 0.1
+
+
+class TestClusteringResultDict:
```
Comment (KennethEnevoldsen) [inline tests/test_abstasks/test_clustering_metrics.py]: remove class (no need - feel free to add a docstring))

File: tests/test_abstasks/test_clustering_metrics.py
Comment (KennethEnevoldsen) [inline tests/test_abstasks/test_clustering_metrics.py:1]: I am not sure we need all of these tests - I suggest that we need to buttom two

File: tests/test_abstasks/test_clustering_metrics.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        )
+
+        assert set(scores) == {"v_measure", "ami"}
+        assert set(scores["v_measure"]) == {"Level 0"}
+        assert set(scores["ami"]) == {"Level 0"}
+        assert len(scores["v_measure"]["Level 0"]) == 3
+        assert len(scores["ami"]["Level 0"]) == 3
+        assert set(assignments) == {"Level 0"}
+
+    def test_perfect_clustering_scores_near_one(self):  # noqa: PLR6301
```
Comment (KennethEnevoldsen) [inline tests/test_abstasks/test_clustering_metrics.py]: I think we trust sklearn here - would delete

File: tests/test_abstasks/test_clustering_metrics.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    labels: list[list[str]] = []
+    for cluster_idx, center in enumerate(centers):
+        for _ in range(points_per_cluster):
+            embeddings.append(center + rng.normal(scale=0.01, size=dim))
+            labels.append([str(cluster_idx)])
+    return np.asarray(embeddings), labels
+
+
+class TestClusteringBootstrap:
+    def test_returns_both_v_measure_and_ami(self):  # noqa: PLR6301
```
Comment (KennethEnevoldsen) [inline tests/test_abstasks/test_clustering_metrics.py]: I would remove

**Review by KennethEnevoldsen — APPROVED** [review]

**Review by KennethEnevoldsen — APPROVED** [review]
Lookgs good - simplified tests a bit
